autoendpoint/src/routers/apns/router.rs

```diff
@@ -125,27 +126,48 @@ impl ApnsRouter {
     async fn handle_error(&self, error: a2::Error, uaid: Uuid, channel: &str) -> ApiError {
         match &error {
             a2::Error::ResponseError(response) => {
+                // capture the APNs error as a metric response. This allows us to spot trends.
+                // While APNS can return a number of errors (see a2::response::ErrorReason) we
+                // shouldn't encounter many of those.
+                let reason = response
```

Comment:
`ErrorReason::to_string` won't work for tags, [it returns descriptive values](https://docs.rs/a2/0.6.2/src/a2/response.rs.html#138) e.g. "The collapse identifier exceeds the maximum allowed size."

Reply:
I'm kind of ok with that, since each string is unique, unless there's a tag value length limit we should avoid.

Reply:
I believe the white space could be problematic. That enum's current `Debug` returns the variant name, I think that might be a better choice despite relying on a `Debug`. E.g. `format!("{:?}", a2::response::ErrorReason::BadCollapseId)` -> "BadCollapseId"